### ISSUE TITLE
Batch vs. sequential inference give different hidden states (diff starts at first layer, ~1e-3–1e-4)

### ISSUE THREAD

**Issue by tengwang0318** [QwenLM/Qwen2.5-VL#1430]
Hey! I’m seeing a mismatch in a VLM during inference.
When I run images one by one, hidden states are stable. If I batch the same images and run in parallel, the hidden states are different — the gap shows up from the first layer, about 1e-3 ~ 1e-4. Padding masks are applied. Even repeating the same image 3× (batch vs. 3 sequential runs) still gives a mismatch.

Expected: same hidden states for the same inputs, regardless of batching.
Actual: different hidden states from layer 1 (≈1e-3–1e-4).

Is this expected (kernel/precision stuff)? Any tips to force batch ≡ sequential / deterministic settings? Thanks!

**Comment by Winsleo** [comment]
I encountered the same problem. My code is as follows:
```python
def demo_vision_language() -> None:
    import torch
    # Replace with your local Qwen2.5-VL path or use the hub id
    model_id = 'Qwen/Qwen2.5-VL-3B-Instruct'
    model = Qwen2_5_VLForConditionalGeneration.from_pretrained(
        model_id,
        torch_dtype=torch.bfloat16,
        device_map='cpu',
    )
    hf_processor = AutoProcessor.from_pretrained(
        model_id, 
        padding_side='left'
    )
    processor = Qwen25VLProcessor(hf_processor)
    
    # Create a pure white PIL Image(224x224)
    white_image = Image.new('RGB', (224, 224), color='white')
    messages1 = [
        {'role': 'system', 'content': 'You are a helpful assistant.'},
        {
            'role': 'user',
            'content': [
                {'type': 'text', 'text': 'What is the color of the image?'},
                # Just as a placeholder
                {'type': 'image',
                 'image': white_image},
            ],
        },
    ]
    
    messages2 = [
        {'role': 'system', 'content': 'You are a helpful assistant.'},
        {
            'role': 'user',
            'content': [
                {'type': 'text', 'text': 'What is in the image?'},
                # You can provide a URL or PIL.Image or path string
                {
                    'type': 'image',
                    'image': "https://qianwen-res.oss-cn-beijing.aliyuncs.com/Qwen-VL/assets/demo.jpeg",
                },
            ],
        },
    ]
    batch_messages = [messages1, messages2]
    text_batch = [
        hf_processor.apply_chat_template(msg, tokenize=False, add_generation_prompt=True)
        for msg in batch_messages
    ]
    text_single = text_batch[0]
    
    from qwen_vl_utils import process_vision_info
    image_inputs_batch, video_inputs_batch = process_vision_info(batch_messages)
    image_inputs_single, video_inputs_single = process_vision_info(messages1)

    batch_inputs = hf_processor(text=text_batch, images=image_inputs_batch, videos=video_inputs_batch, padding=True, return_tensors='pt')
    single_inputs = hf_processor(text=text_single, images=image_inputs_single, videos=video_inputs_single, padding=True, return_tensors='pt')

    batch_generated_ids = model.generate(**batch_inputs, max_new_tokens=128)
    generated_ids_trimmed = [
        out_ids[len(in_ids) :] for in_ids, out_ids in zip(batch_inputs.input_ids, batch_generated_ids)
    ]
    single_generated_ids = model.generate(**single_inputs, max_new_tokens=128)
    single_generated_ids_trimmed = [
        out_ids[len(in_ids) :] for in_ids, out_ids in zip(single_inputs.input_ids, single_generated_ids)
    ]
    texts = hf_processor.batch_decode(generated_ids_trimmed)
    single_text = hf_processor.batch_decode(single_generated_ids_trimmed)
    print('batch output:')
    print(texts[0])
    print()
    print('single output:')
    print(single_text[0])
```
There are big differences between batch inference mode and single inference mode:
<img width="1485" height="279" alt="Image" src="https://github.com/user-attachments/assets/4efcba44-56b6-4c6c-b86a-28a9ab31bb01" />

**Comment by tengwang0318** [comment]
but the issue hasn't been addressed? Why do you close this issue?

**Comment by KpKqwq** [comment]
any update？
